### PR TITLE
Release 0.15.0

### PR DESCRIPTION
**Pull request by hagenw**
My try on summarizing our big chunk of work:

image

### CODE REVIEW

**Comment by hagenw** [comment]
It's still marked as WIP as we have other open pull requests, but feel free to review already the text of the changelog.

**Comment by hagenw** [comment]
All needed pull requests are now merged and we are ready for the release.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -7,6 +7,74 @@ The format is based on `Keep a Changelog`_,
 and this project adheres to `Semantic Versioning`_.
 
 
+Version 0.15.0 (2022-08-05)
+---------------------------
+
+* Added: ``audformat.MiscTable``
+  which can store labels
```
Comment (frankenjoe) [inline CHANGELOG.rst:14]: I would even use a more generic here:

```suggestion
  which can store data
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -7,6 +7,74 @@ The format is based on `Keep a Changelog`_,
...
+  not associated with media files
+* Added: usage of a misc table as labels in a scheme
```
Comment (frankenjoe) [inline CHANGELOG.rst:16]:
```suggestion
* Added: store scheme labels in a misc table
```

File: CHANGELOG.rst [hunk 3]
```diff
@@ -7,6 +7,74 @@ The format is based on `Keep a Changelog`_,
...
+* Added: ``audformat.utils.set_index_dtypes()``
+  to change dtypes of an index
+* Added: ``audformat.testing.add_misc_table()``
+* Changed: ``audformat.Database.update()``
+  can now join schemes
+  with different labels
+* Changed: ``audformat.utils.union()``
+  and ``audformat.utils.intersect()``
+  now support any kind of index
+* Changed: ``audformat.utils.concat()``
```
Comment (frankenjoe) [inline CHANGELOG.rst:38]: I would also add this to the previous bullet

File: CHANGELOG.rst [hunk 4]
```diff
@@ -7,6 +7,74 @@ The format is based on `Keep a Changelog`_,
...
+  can now concatenate objects
+  that contain any kind of index
+* Changed: ``audformat.Database.__iter__``
+  iterates through all (misc) tables,
+  e.g. a user can do ``list(db)``
```
Comment (frankenjoe) [inline CHANGELOG.rst:43]:
```suggestion
  e.g. a user can do ``list(db)``
  to get a list of all (misc) tables
```

File: CHANGELOG.rst [hunk 5]
```diff
@@ -7,6 +7,74 @@ The format is based on `Keep a Changelog`_,
...
+* Changed: require ``pandas>=1.4.1``
```
Comment (frankenjoe) [inline CHANGELOG.rst:44]: Should we mention that this also requires `Python >= 3.7`?

Comment (hagenw) [reply CHANGELOG.rst:44]: It requires `Python >= 3.8`.
So far I decided to just mention at the end, that we removed support for Python 3.7 without any reason

File: CHANGELOG.rst [hunk 6]
```diff
@@ -7,6 +7,74 @@ The format is based on `Keep a Changelog`_,
...
+* Changed: use ``pandas`` dtype ``'string'``
+  instead of ``'object'``
+  for storing ``audformat`` dtype ``'str'`` entries
+* Changed: use a misc table
+  to store the ``'speaker'`` scheme labels
+  in the emodb example
+  in the documentation
+* Fixed: ensure column IDs are different from index level names
+* Fixed: make sure incoming data
```
Comment (frankenjoe) [inline CHANGELOG.rst:52]: I think we should mention that this applies to `Column.set()` so maybe:

```
Column.set()
converts data to dtype of scheme
before checking if values 
are in range
```

File: CHANGELOG.rst (hunk 3 above)
Comment (hagenw) [reply CHANGELOG.rst]: I changed it accordingly.

File: CHANGELOG.rst [hunk 7]
```diff
@@ -7,6 +7,74 @@ The format is based on `Keep a Changelog`_,
...
+  ``audformat.Column``,
+  ``audformat.Database``,
+  ``audformat.Media``,
+  ``audformat.Rater``,
+  ``audformat.Scheme``,
+  ``audformat.Split``,
+  ``audformat.Table``
+* Fixed: type hint of argument ``dtype``
+  in the documentation of ``audformat.Scheme``
+* Removed: support for Python 3.7
```
Comment (frankenjoe) [inline CHANGELOG.rst:74]: Ah ok, you mention it down here :)

File: CHANGELOG.rst [hunk 8]
```diff
@@ -7,6 +7,74 @@ The format is based on `Keep a Changelog`_,
...
+* Changed: use a misc table
+  to store the ``'speaker'`` scheme labels
+  in the emodb example
+  in the documentation
+* Fixed: ensure column IDs are different from index level names
+* Fixed: make sure incoming data
+  is converted to matching dtype of scheme
+  before checking if the value lies in the max-min-range
+  of the scheme
+* Fixed: links to ``pandas`` objects in the documentation
```
Comment (frankenjoe) [inline CHANGELOG.rst:56]:
```suggestion
* Fixed: links to ``pandas`` API in the documentation
```

**Comment by frankenjoe** [comment]
Maybe also add:

```
* Changed: audformat.utils.join_labels()`
  raises ``ValueError`` if labels are of
  different dtype 
```

File: CHANGELOG.rst (hunk 6 above)
Comment (hagenw) [reply CHANGELOG.rst]: I mention `Column.set()` now here

**Comment by hagenw** [comment]
> different dtype

Good idea, I added it as well

File: CHANGELOG.rst [hunk 9]
```diff
@@ -7,6 +7,77 @@ The format is based on `Keep a Changelog`_,
...
+  to change dtypes of an index
+* Added: ``audformat.testing.add_misc_table()``
+* Changed: ``audformat.Database.update()``
+  can now join schemes
+  with different labels
+* Changed: ``audformat.utils.union()``,
+  ``audformat.utils.intersect()``,
+  and ``audformat.utils.concat()``
+  now support any kind of index
+* Changed: ``audformat.Database.__iter__``
```
Comment (frankenjoe) [inline CHANGELOG.rst:39]: I think we added this feature, so it should be:

```suggestion
* Added: ``audformat.Database.__iter__``
```

File: CHANGELOG.rst [hunk 10]
```diff
@@ -7,6 +7,77 @@ The format is based on `Keep a Changelog`_,
...
+* Added: ``audformat.utils.set_index_dtypes()``
+  to change dtypes of an index
+* Added: ``audformat.testing.add_misc_table()``
+* Changed: ``audformat.Database.update()``
+  can now join schemes
+  with different labels
+* Changed: ``audformat.utils.union()``,
+  ``audformat.utils.intersect()``,
+  and ``audformat.utils.concat()``
+  now support any kind of index
```
Comment (frankenjoe) [inline CHANGELOG.rst:42]: I think we also need to mention that we convert any integer type to `Int64`

File: CHANGELOG.rst (hunk 9 above)
Comment (hagenw) [reply CHANGELOG.rst]: Yes, that's correct.
I added it in a commit to also move it upwards.

File: CHANGELOG.rst (hunk 10 above)
Comment (hagenw) [reply CHANGELOG.rst:42]: This is not really a change.
Before integer dtype was not supported.

Comment (frankenjoe) [reply CHANGELOG.rst:42]: Ah yes, that's true. But what we dropped is the special treatment when providing a mix of `filewise` and `segmented` index objects.

**Comment by frankenjoe** [comment]
Cool, except for that last conversation that is still open we are good to go I think!

File: CHANGELOG.rst (hunk 10 above)
Comment (hagenw) [reply CHANGELOG.rst:42]: Good point, I tried to summarize this in an additional point now with:

```rst
* Changed: ``audformat.utils.intersect()``
  no longer removes segments
  from a segmented index
  that are contained
  in a filewise index
```

**Pull request merged**